server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -1126,4 +1137,50 @@ public void modeChanged(Mode mode) {
             }
         }
     }
+
+    /**
+     * Lookup job status by job id.
+     *
+     * @param jobId
+     *  Id of the requested job
+     *
+     * @return
+     *  {@link AsyncJobStatus} of the requested job
+     *
+     * @throws JobAccessException.NotFound
+     *  When requested job was not found
+     */
+    public AsyncJobStatus get(final String jobId) throws JobAccessException {
+        final AsyncJobStatus status = this.jobCurator.get(jobId);
+        if (status == null) {
+            throw new JobAccessException.NotFound(jobId);
+        }
+        return status;
+    }
+
+    /**
+     * Request cancellation of the job with the given id.
+     *
+     * @param jobId
+     *  Id of the cancelled job
+     *
+     * @return
+     *  {@link AsyncJobStatus} of the cancelled job
+     *
+     * @throws JobAccessException.NotFound
+     *  When requested job was not found
+     *
+     * @throws JobAccessException.InvalidState
+     *  When cancelled job is already terminated (CANCELLED, COMPLETED, ...)
+     */
+    public AsyncJobStatus cancel(final String jobId) throws JobAccessException {
+        final AsyncJobStatus status = this.get(jobId);
+        if (status.getState().isTerminal()) {
+            throw new JobAccessException.InvalidState(jobId, status.getState());
+        }
+        status.setState(JobState.CANCELED);
+        this.jobCurator.saveAll(Collections.singletonList(status), false, false);
```

Comment:
jobCurator.merge works here without needing to wrap the object in a list.

Reply:
Done.